lib/BackgroundJob/BackgroundService.php

```diff
@@ -95,25 +95,34 @@ public function execute(int $timeout, bool $verbose, IUser $user = null) {
 			UnlockTask::class
 		];
 
-		// todo: implement bailing with exceptions
 		// Main logic to iterate over all tasks and executes them.
 		//
 		$startTime = time();
-		for ($i=0; $i < count($task_classes); $i++) {
+		for ($i=0, $task_classes_count = count($task_classes); $i < $task_classes_count; $i++) {
 			$task_class = $task_classes[$i];
 			$task = $this->application->getContainer()->query($task_class);
 			$this->context->logger->logInfo(sprintf("%d/%d - Executing task %s (%s)",
 				$i+1, count($task_classes), (new \ReflectionClass($task_class))->getShortName(), $task->description()));
 
-			$generator = $task->do($this->context);
-			foreach ($generator as $_) {
-				$currentTime = time();
-				if (($timeout > 0) && ($currentTime - $startTime > $timeout)) {
-					$this->context->logger->logInfo("Time out. Quitting...");
-					return;
-				}
+			try {
+				$generator = $task->do($this->context);
+				foreach ($generator as $_) {
+					$currentTime = time();
+					if (($timeout > 0) && ($currentTime - $startTime > $timeout)) {
+						$this->context->logger->logInfo("Time out. Quitting...");
+						return;
+					}
 
-				$this->context->logger->logDebug('yielding');
+					$this->context->logger->logDebug('yielding');
+				}
+			} catch (\Exception $e) {
+				// Any exception is fatal, and we should quit background job
+				//
```

Comment:
Handle exception here is correct, but should not fail when control a simple dependency, since it is a routine control...